esrally/driver/runner.py

```diff
@@ -2464,18 +2464,17 @@ async def __call__(self, es, params):
         pages -= 1
         weight = 1
 
-        cursor = parse(r, ["cursor"]).get("cursor")
+        while pages > 0:
+            cursor = parse(r, ["cursor"]).get("cursor")
+
+            if not cursor:
+                raise exceptions.DataError(
+                    "Result set has been exhausted before all pages have been fetched, {} page(s) remaining.".format(pages)
+                )
 
-        while cursor and pages > 0:
             r = await es.transport.perform_request("POST", "/_sql", body={"cursor": cursor})
```

Comment:
Should we clarify in the [docs](https://esrally.readthedocs.io/en/latest/track.html#sql) for this runner that when pages > 1 and the response is large, json parsing can be expensive and affect results (latency)?

Reply:
good point! I added a note to the docs.